### PR TITLE
Specify Java version; Load default Java version;

### PR DESCRIPTION
**Pull request by osmarcf**
Pulling a "Specify Java version" feature.
It is simple to change the Java version when the project is created (issue 136 is closed with that comment - https://github.com/microsoft/vscode-spring-initializr/issues/136 ), but I added this feature since VSCode is turning to use Java 11, so it is helpful to choose Java version when creating the project.

### CODE REVIEW

**Comment by Eskibear** [comment]
Thanks for the contribution. 
Basically we only allow to specify the most important properties, in order to decrease the number of steps (as currently we cannot go backward among steps). You can always change below properties in pom.xml after the project is created.
- **Java Version**: As Java on VS Code now requires JDK 11 to run, I agree that we should allow users to specify it.  **Suggest to change the default value to "11".** 
- **Name**: By default it's the same as "artifact id". Note that in start.spring.io, once you change the artifact id, the name is also changed. So we follow the behavior. **Suggest to remove this setting.** 

BTW we're switching to consume the [latest (v2.2) metadata](https://docs.spring.io/initializr/docs/current-SNAPSHOT/reference/html/#api-guide). **Ideally** we don't have to have the settings to hardcode the options, as they can be retrieved by querying the service.

**Comment by osmarcf** [comment]
> Thanks for the contribution.

Thanks for reviewing!

> Basically we only allow to specify the most important properties, in order to decrease the number of steps (as currently we cannot go backward among steps). You can always change below properties in pom.xml after the project is created.

I see that to decrease the steps, the user should set default values. The user change the pom.xml is always a choice, but I also see that this extension should replace the need for it (and even the access the start.spring.io)

> 
>     * **Java Version**: As Java on VS Code now requires JDK 11 to run, I agree that we should allow users to specify it.  **Suggest to change the default value to "11".**

Fine! I will commit that later!

> 
>     * **Name**: By default it's the same as "artifact id". Note that in start.spring.io, once you change the artifact id, the name is also changed. So we follow the behavior. **Suggest to remove this setting.**

I have a comment about setting the "Name":  in the current version, changing the ArtifactId change the pom.xml file, but it doesn't affect the Application Name, that is, the "main" class will have the default name "DemoApplication" (and also the filename DemoApplication.java). 


> 
> 
> BTW we're switching to consume the [latest (v2.2) metadata](https://docs.spring.io/initializr/docs/current-SNAPSHOT/reference/html/#api-guide). **Ideally** we don't have to have the settings to hardcode the options, as they can be retrieved by querying the service.

Nice!

File: package.json [hunk 1]
```diff
@@ -108,6 +120,12 @@
           "scope": "window",
           "description": "Default value for Artifact Id."
         },
+        "spring.initializr.defaultName": {
```
Comment (Eskibear) [inline package.json:123]: we don't have to have this setting

File: README.md [hunk 2]
```diff
@@ -44,9 +44,15 @@ ext install vscode-spring-initializr
   // Default language.
   "spring.initializr.defaultLanguage": "Java",
 
+  // Default Java version.
+  "spring.initializr.defaultJavaVersion": "11",
+
   // Default value for Artifact Id.
   "spring.initializr.defaultArtifactId": "demo",
 
+  // Default value for Name.
```
Comment (Eskibear) [inline README.md:53]: we don't need it

File: TestPlan.md [hunk 3]
```diff
@@ -64,6 +64,13 @@ After that, verify `build.gradle` instead of `pom.xml`.
 1. Verify:
     1. `GroupId` and `ArtifactId` are filled by the specified default values.
 
+### Default value of Name.
```
Comment (Eskibear) [inline TestPlan.md:67]: remove this section

File: src/Utils/index.ts [hunk 4]
```diff
@@ -114,6 +114,10 @@ export function artifactIdValidation(value: string): string {
     return (/^[a-z_][a-z0-9_]*(-[a-z_][a-z0-9_]*)*$/.test(value)) ? null : "Invalid Artifact Id";
 }
 
+export function nameValidation(value: string): string {
```
Comment (Eskibear) [inline src/Utils/index.ts:117]: remove this

File: src/handler/GenerateProjectHandler.ts [hunk 5]
```diff
@@ -132,6 +155,16 @@ async function specifyArtifactId(): Promise<string> {
     });
 }
 
+async function specifyName(): Promise<string> {
+    const defaultName: string = vscode.workspace.getConfiguration("spring.initializr").get<string>("defaultName");
+    return await getFromInputBox({
+        placeHolder: "e.g. demo (will turn into DemoApplication class - mostly the same as ArtifactId)",
+        prompt: "Input project name.",
+        validateInput: nameValidation,
+        value: defaultName,
```
Comment (Eskibear) [inline src/handler/GenerateProjectHandler.ts:164]: As you mentioned: 
>  mostly the same as ArtifactId

Let's remove this step, just re-use the value of artifactId.

File: src/handler/GenerateProjectHandler.ts [hunk 6]
```diff
@@ -90,8 +100,10 @@ export class GenerateProjectHandler extends BaseHandler {
         const params: string[] = [
             `type=${this.projectType}`,
             `language=${this.language}`,
+            `javaVersion=${this.javaVersion}`,
             `groupId=${this.groupId}`,
             `artifactId=${this.artifactId}`,
+            `name=${this.name}`,
```
Comment (Eskibear) [inline src/handler/GenerateProjectHandler.ts:106]: `name=${this.artifactId}`

Comment (osmarcf) [reply src/handler/GenerateProjectHandler.ts]: Done.

File: src/handler/GenerateProjectHandler.ts (hunk 5 above)
Comment (osmarcf) [reply src/handler/GenerateProjectHandler.ts]: Done.

File: src/Utils/index.ts (hunk 4 above)
Comment (osmarcf) [reply src/Utils/index.ts]: Done.

File: TestPlan.md (hunk 3 above)
Comment (osmarcf) [reply TestPlan.md]: Done.

File: README.md (hunk 2 above)
Comment (osmarcf) [reply README.md]: Done.

File: package.json (hunk 1 above)
Comment (osmarcf) [reply package.json]: Done.

**Pull request merged**